tests/cli-paths.rs

```diff
@@ -309,21 +309,69 @@ mod windows {
     use crate::mock::clitools::{self, expect_ok, Scenario};
 
     #[test]
-    #[cfg(windows)]
     /// Smoke test for end-to-end code connectivity of the installer path mgmt on windows.
     fn install_uninstall_affect_path() {
         clitools::setup(Scenario::Empty, &|config| {
             with_saved_path(&|| {
-                let path = config.cargodir.join("bin").to_string_lossy().to_string();
+                let path = format!("{:?}", config.cargodir.join("bin").to_string_lossy());
 
                 expect_ok(config, &INIT_NONE);
                 assert!(
-                    get_path().unwrap().contains(&path),
-                    format!("`{}` not in `{}`", get_path().unwrap(), &path)
+                    get_path()
+                        .unwrap()
+                        .unwrap()
+                        .to_string()
+                        .contains(path.trim_matches('"')),
+                    format!("`{}` not in `{}`", path, get_path().unwrap().unwrap())
                 );
 
                 expect_ok(config, &["rustup", "self", "uninstall", "-y"]);
-                assert!(!get_path().unwrap().contains(&path));
+                assert!(!get_path().unwrap().unwrap().to_string().contains(&path));
+            })
+        });
+    }
+
+    #[test]
+    /// Smoke test for end-to-end code connectivity of the installer path mgmt on windows.
+    fn install_uninstall_affect_path_with_non_unicode() {
```

Comment:
thank you. I would have actually been ok with no functional test given the excellent local coverage we have now for this part of the system, but :shrug: